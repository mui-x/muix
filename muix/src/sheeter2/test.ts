import { compile } from './compiler2'
import { TSheeterSource, TSheeterCompiled } from './types'

const merging: TSheeterSource.RulesTreeRoot<'root' | 'disabled'> = {
    $before: {
        before: 'before',
        self: 'before',
        web: 'before',
        after: 'before',
    },
    self: 'self',
    web: 'self',
    after: 'self',
    $web: {
        web: 'web',
        after: 'web',
    },
    $after: {
        after: 'after',
    },
}

const whenUsed: TSheeterSource.RulesTreeRoot<'root' | 'disabled'> = {
    $whenUsed: {
        root: {
            plain: 'plain',
        }
    },
    ':hover': {
        $whenUsed: {
            root: {
                hover: 'hover',
            }
        },
    }
}

const whenUsedRecursion: TSheeterSource.RulesTreeRoot<'root' | 'disabled'> = {
    rule: 'rule',
    $whenUsed: {
        root: {
            plainRoot: 'plainRoot',
            $whenUsed: {
                disabled: {
                    plainInner: 'plainInner',
                    ':hover': {
                        plainInnerHover: 'plainInnerHover',
                        ':active': {
                            plainInnerHoverActive: 'plainInnerHoverActive',
                        }
                    }
                }
            },
        }
    },
    ':hover': {
        hover: 'hover',
        $whenUsed: {
            root: {
                hoverRoot: 'hoverRoot',
                $whenUsed: {
                    disabled: {
                        hoverRootInner: 'hoverRootInner',
                    }
                },
                ':active': {
                    hoverRootActive: 'hoverRootActive',
                    $whenUsed: {
                        disabled: {
                            hoverRootActiveInner: 'hoverRootActiveInner',
                        }
                    },
                },
            }
        },
    }
}



export const run = () => {
    let res
    //res = JSON.stringify(compile(merging, 'root'), null, 2)
    //res = JSON.stringify(compile(whenUsed, 'root'), null, 2)
    res = JSON.stringify(compile(whenUsedRecursion, 'root'), null, 2)
    debugger
}

