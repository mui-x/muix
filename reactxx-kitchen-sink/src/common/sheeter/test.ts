
import * as Sheeter from 'reactxx-sheeter'

export const test = () => {

  const patchable = Sheeter.toPatchableAndMergeable(root)

  const finishAddIns: Sheeter.FinishAddIns = {
    $mediaq: Sheeter.mediaqFinishAddInCreator()
  }
  const addInRulesetFilters: Sheeter.RulesetPatchGetters = {
    $mediaq: Sheeter.mediaqRulesetPatchGetterCreator(200), // screen width = 200px
    $whenUsed: Sheeter.whenUsedRulesetFilter
  }
  window.isWeb = true
  const mergedWeb = Sheeter.mergeSheetsAndFinish(patchable, [], finishAddIns)
  const codeRootWeb = Sheeter.mergeRulesetsForCode(mergedWeb, addInRulesetFilters, [mergedWeb.root, false as any, mergedWeb.b, null as any])

  window.isWeb = false
  const mergedNative = Sheeter.mergeSheetsAndFinish(patchable, [], finishAddIns)
  const codeRootNative = Sheeter.mergeRulesetsForCode(mergedNative, addInRulesetFilters, [mergedNative.root, false as any, mergedNative.b, null as any])

  debugger
}

//****************************
// INPUT DATA

const root: Sheeter.Sheet = {
  root: {
    a0: 1,
    $web: {
      a1: 1,
      ":active": {
        a2: 1,
        $before: {
          $mediaq: {
            '100-400': {
              a3: 1,
              $whenUsed: {
                b: {
                  a4: 1,
                  $after: {
                    $web: {
                      a5: 1,
                      ":hover": {
                        a6: 1,
                        $mediaq: {
                          '200-300': {
                            a7: 1,
                          }
                        } as Sheeter.Sheet
                      },
                    },
                    ":hover": {
                      $mediaq: {
                        '300-400': {
                          a8: 1,
                        }
                      } as Sheeter.Sheet
                    }
                  },
                },
              } as Sheeter.Sheet,
            },
          } as Sheeter.Sheet,
        },
      },
    },
  },
  b: {}
}

//window['isWeb'] = true

//****************************
// 

const _toPatchableAndMergeable = {
  "root": {
    "a0": 1,
    "a1": 1,
    ":active": {
      "a2": 1
    }
  },
  "b": {},
  "$addInsFinal": {
    "$mediaq": {
    }
  },
  "$addIns": {
    "$mediaq": {
      "$addIns/$whenUsed/$addIns/$mediaq/root/:active/100-200/b/:hover": {
        "300-400": {
          "a8": 1
        },
        "200-300": {
          "a7": 1
        },
        "#path": [
          "$addIns",
          "$whenUsed",
          "$addIns/$mediaq/root/:active/100-200",
          "b",
          ":hover"
        ],
      },
      "root/:active": {
        "100-200": {
          "a3": 1
        },
        "#path": [
          "root",
          ":active"
        ]
      }
    },
    "$whenUsed": {
      "$addIns/$mediaq/root/:active/100-200": {
        "b": {
          "a4": 1,
          ":hover": {
            "a6": 1
          },
          "a5": 1
        },
        "#path": [
          "$addIns",
          "$mediaq",
          "root/:active",
          "100-200"
        ]
      }
    }
  }
}

const inGetPatchLow = {
  "rootPatches": [
    {
      "path": [
        "root",
        ":active"
      ],
      "items": [
        {
          "a3": 1,
          "a4": 1,
          ":hover": {
            "a6": 1,
            "a8": 1,
            "a7": 1
          },
          "a5": 1
        }
      ]
    }
  ],
  "addInPatches": [
    {
      "path": [
        "$addIns",
        "$whenUsed",
        "$addIns/$mediaq/root/:active/100-200",
        "b",
        ":hover"
      ],
      "items": [
        {
          "a8": 1
        },
        {
          "a7": 1
        }
      ]
    },
    {
      "path": [
        "$addIns",
        "$mediaq",
        "root/:active",
        "100-200"
      ],
      "items": [
        {
          "a4": 1,
          ":hover": {
            "a6": 1,
            "a8": 1,
            "a7": 1
          },
          "a5": 1
        }
      ]
    }
  ]
}

const result = {
  "a0": 1,
  "a1": 1,
  ":active": {
    "a2": 1,
    "a3": 1,
    "a4": 1,
    ":hover": {
      "a6": 1,
      "a8": 1,
      "a7": 1
    },
    "a5": 1
  }
}