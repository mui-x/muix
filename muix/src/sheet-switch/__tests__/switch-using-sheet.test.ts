import { atomizeSheet } from "reactxx-sheeter";
import { initPlatform, toClassNames, Shape } from "reactxx-tests";
import { createSheet, query } from "./switch-sheet.test";

describe("SWITCH using sheet", () => {
  it("NATIVE", () => {
    initPlatform(false);

    const sheet = atomizeSheet<Shape>(createSheet());

    toClassNames(
      "NATIVE, OPENED",
      query(true),
      [sheet.root, sheet.nativeOnly],
      res =>
        res.toMatchSnapshot()
    );

    toClassNames(
      "NATIVE, CLOSED",
      query(false),
      [sheet.root, sheet.nativeOnly],
      res =>
        res.toMatchSnapshot()
    );
  });
  it("WEB", () => {
    initPlatform(true);

    const sheet = atomizeSheet<Shape>(createSheet());

    toClassNames("WEB, OPENED", query(true), [sheet.root, sheet.webOnly], res =>
      res.toMatchSnapshot()
    );

    toClassNames(
      "WEB, CLOSED",
      query(false),
      [sheet.root, sheet.webOnly],
      res =>
        res.toMatchSnapshot()
    );
  });
});
