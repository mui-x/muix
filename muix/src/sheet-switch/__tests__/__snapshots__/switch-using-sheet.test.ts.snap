// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SWITCH define sheet NATIVE 1`] = `
Object {
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "backgroundColor: red /*sheet.root/$switch.isOpened*/",
        ],
        "conditions": Array [
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          "borderColor: cyan /*sheet.root/$switch.isOpened/$switch.isOpened*/",
        ],
        "conditions": Array [
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          "backgroundColor: brown /*sheet.root/$native/$switch.isOpened*/",
        ],
        "conditions": Array [
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`;

exports[`SWITCH define sheet WEB 1`] = `
Object {
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".a { background-color:red /*sheet.root/$switch.isOpened*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          ".b { border-color:cyan /*sheet.root/$switch.isOpened/$switch.isOpened*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          ".c:hover { background-color:green /*sheet.root/$web/:hover/$switch.isClosed[0]*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
  "webOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".d:hover { background-color:yellow /*sheet.webOnly/$web/:hover/$switch.isOpened*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
    ],
    "name": "webOnly",
    "~": "c",
  },
}
`;

exports[`SWITCH using sheet NATIVE 1`] = `
Array [
  "
##########################################
##########################################
#
#  NATIVE, OPENED
#
##########################################
##########################################
",
  "
******************************************
*  QUERY TO STHEET
******************************************
",
  Array [
    "backgroundColor: red /*sheet.root/$switch.isOpened*/",
    "borderColor: cyan /*sheet.root/$switch.isOpened/$switch.isOpened*/",
    "backgroundColor: brown /*sheet.root/$native/$switch.isOpened*/",
  ],
  "
******************************************
*  APPLY LASTWINS STRATEGY
******************************************
",
  "
backgroundColor: brown /*sheet.root/$native/$switch.isOpened*/
borderColor: cyan /*sheet.root/$switch.isOpened/$switch.isOpened*/",
  "
******************************************
*  FINALIZE FOR for web's className or native's style
******************************************
",
  Object {
    "backgroundColor": "brown",
    "borderColor": "cyan",
  },
]
`;

exports[`SWITCH using sheet NATIVE 2`] = `
Array [
  "
##########################################
##########################################
#
#  NATIVE, CLOSED
#
##########################################
##########################################
",
  "
******************************************
*  QUERY TO STHEET
******************************************
",
  Array [],
  "
******************************************
*  APPLY LASTWINS STRATEGY
******************************************
",
  "",
  "
******************************************
*  FINALIZE FOR for web's className or native's style
******************************************
",
  Object {},
]
`;

exports[`SWITCH using sheet WEB 1`] = `
Array [
  "
##########################################
##########################################
#
#  WEB, OPENED
#
##########################################
##########################################
",
  "
******************************************
*  QUERY TO STHEET
******************************************
",
  "
.a { background-color:red /*sheet.root/$switch.isOpened*/ }
.b { border-color:cyan /*sheet.root/$switch.isOpened/$switch.isOpened*/ }
.d:hover { background-color:yellow /*sheet.webOnly/$web/:hover/$switch.isOpened*/ }",
  "
******************************************
*  APPLY LASTWINS STRATEGY
******************************************
",
  "
.d:hover { background-color:yellow /*sheet.webOnly/$web/:hover/$switch.isOpened*/ }
.b { border-color:cyan /*sheet.root/$switch.isOpened/$switch.isOpened*/ }
.a { background-color:red /*sheet.root/$switch.isOpened*/ }",
  "
******************************************
*  FINALIZE FOR for web's className or native's style
******************************************
",
  "d b a",
]
`;

exports[`SWITCH using sheet WEB 2`] = `
Array [
  "
##########################################
##########################################
#
#  WEB, CLOSED
#
##########################################
##########################################
",
  "
******************************************
*  QUERY TO STHEET
******************************************
",
  "
.c:hover { background-color:green /*sheet.root/$web/:hover/$switch.isClosed[0]*/ }",
  "
******************************************
*  APPLY LASTWINS STRATEGY
******************************************
",
  "
.c:hover { background-color:green /*sheet.root/$web/:hover/$switch.isClosed[0]*/ }",
  "
******************************************
*  FINALIZE FOR for web's className or native's style
******************************************
",
  "c",
]
`;
